src/test/java/com/google/cloud/spanner/pgadapter/python/Django/DjangoBasicTest.java

```diff
@@ -0,0 +1,191 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.cloud.spanner.pgadapter.python.Django;
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.cloud.spanner.MockSpannerServiceImpl.StatementResult;
+import com.google.cloud.spanner.Statement;
+import com.google.cloud.spanner.pgadapter.python.PythonTest;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.ListValue;
+import com.google.protobuf.Value;
+import com.google.spanner.v1.ExecuteSqlRequest;
+import com.google.spanner.v1.ResultSet;
+import com.google.spanner.v1.ResultSetMetadata;
+import com.google.spanner.v1.StructType;
+import com.google.spanner.v1.StructType.Field;
+import com.google.spanner.v1.Type;
+import com.google.spanner.v1.TypeCode;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+@Category(PythonTest.class)
+public class DjangoBasicTest extends DjangoTestSetup {
+
+  @Parameter public String host;
+
+  @Parameters(name = "host = {0}")
+  public static List<Object[]> data() {
+    return ImmutableList.of(new Object[] {"localhost"}, new Object[] {"/tmp"});
+  }
+
+  private ResultSet createResultSet(List<String> rows) {
```

Comment:
(You can leave this as is for now, but as a tip for further development): This works well for now as the sample is still very small, but I think that as you increase the number of classes (models) and the number of columns, it will be better to create a replica of your model classes in PHP in Java as well, and use that here.

Reply:
Noted. Just one thing, you meant Python instead of PHP, right?